apps/web/src/pages/auth/InvitationPage.tsx

```diff
@@ -11,43 +11,48 @@ import { SignUpForm } from '../../components/auth/SignUpForm';
 import { colors, Text, Button } from '../../design-system';
 import { AuthContext } from '../../store/authContext';
 import { useAcceptInvite } from '../../components/auth/use-accept-invite.hook';
-import { When } from '../../components/utils/When';
+import { LoginForm } from '../../components/auth/LoginForm';
 
 export default function InvitationPage() {
+  const queryClient = useQueryClient();
   const navigate = useNavigate();
   const { token, logout, currentUser } = useContext(AuthContext);
   const location = useLocation();
   const isLoggedIn = !!token;
-  const { token: tokenParam } = useParams<{ token: string }>();
-  const { isLoading: loadingAcceptInvite, submitToken } = useAcceptInvite();
+  const { token: invitationToken } = useParams<{ token: string }>();
+  const tokensRef = useRef({ token, invitationToken });
+  tokensRef.current = { token, invitationToken };
+  const { isLoading: isAcceptingInvite, submitToken } = useAcceptInvite();
   const { data, isInitialLoading } = useQuery<IGetInviteResponseDto, IGetInviteResponseDto>(
     ['getInviteTokenData'],
-    () => getInviteTokenData(tokenParam || ''),
+    () => getInviteTokenData(invitationToken || ''),
     {
-      enabled: !!tokenParam,
+      enabled: !!invitationToken,
+      refetchOnWindowFocus: false,
     }
   );
   const inviterFirstName = data?.inviter?.firstName || '';
   const organizationName = data?.organization.name || '';
-
-  const existingUser = tokenParam && data?._userId;
-  const invalidCurrentUser = existingUser && currentUser && currentUser._id !== data?._userId;
-
-  const acceptToken = async () => {
-    if (existingUser && currentUser && currentUser._id === data?._userId && isLoggedIn) {
-      const result = await submitToken(tokenParam as string, true);
-      if (result) navigate('/templates');
-    }
-  };
+  const existingUser = !!(invitationToken && data?._userId);
+  const isLoggedInAsInvitedUser = !!(isLoggedIn && existingUser && currentUser && currentUser._id === data?._userId);
 
   const logoutWhenActiveSession = () => {
     logout();
     navigate(location.pathname);
   };
 
   useEffect(() => {
-    acceptToken();
-  }, [tokenParam, data, currentUser]);
+    // auto accept invitation when logged in as invited user
+    if (isLoggedInAsInvitedUser) {
+      submitToken(tokensRef.current.token as string, tokensRef.current.invitationToken as string, true);
+    }
+  }, [isLoggedInAsInvitedUser]);
+
+  useEffect(() => {
+    return () => {
+      queryClient.removeQueries(['getInviteTokenData']);
```

Comment:
we have to remove this query, otherwise, the `queryClient.refetch` will refetch it with a failed request.

Reply:
This is a great catch.